R/checkSummations.R

```diff
@@ -105,12 +112,26 @@ checkSummations <- function(mifFile, outputDirectory = ".", template = NULL, sum
     # calculate differences for comparison
     comp <- comp %>%
       group_by(!!!syms(c("model", "scenario", "region", "period", "variable", "unit", "value"))) %>%
-      summarise(checkSum = sum(!!sym("childVal") * !!sym("factor")), .groups = "drop") %>%
+      summarise(checkSum = sum(!!sym("childVal") * !!sym("factor"), na.rm = TRUE),
+        summation = paste(
+          ifelse(
+            is.na(!!sym("child")),
+            "",
+            ifelse(
+              !!sym("factor") == 1,
+              paste0(!!sym("child"), " (", !!sym("childVal"), ")"),
+              paste0(!!sym("factor"), " * ", !!sym("child"), " (", !!sym("childVal"), ")")
+              )
+            ),
+          collapse = " + "),
+        .groups = "drop") %>%
       ungroup() %>%
       mutate(
         diff = !!sym("checkSum") - !!sym("value"),
-        reldiff = 100 * (!!sym("checkSum") - !!sym("value")) / !!sym("value")
-      )
+        reldiff = round(100 * (!!sym("checkSum") - !!sym("value")) / !!sym("value"), 2),
```

Comment:
are you sure this is a good idea? This can lead to basically have `reldiff = 0` in the data. Maybe use `niceround()` which always keeps some significant digits…

Reply:
commit https://github.com/pik-piam/piamInterfaces/pull/186/commits/5500bd7462d2492245709a6056f05afec78d4af2 moves the rounding to only affect the csv file, and not the inner value of the relative difference.

I couldn't use `niceround` instead of round, as the line trows an error instead:
```
Error in `arrange()`:
ℹ In argument: `..1 = abs(reldiff)`.
Caused by error in `abs()`:
! non-numeric argument to mathematical function 
```
The function probably cannot handle some non numerical value.

The rounding was a request from Robert and it should only round to zero in the csv file values that have relative difference lower than 0.005 %. I would say that this is small enough to be disconsidered.

Reply:
I think I disagree. We use this function also for our internal REMIND checks and here we are much more sensitive. Have you tried putting the `niceround` after the part with the `abs` function, so 
```
arrange(tmp, desc(abs(!!sym("reldiff")))) %>% mutate(reldiff = niceround(reldiff)),
```

Reply:
@orichters could you briefly explain what for you would be using the relative difference value with very small values? I mean if you just want to get all things correct, you still have the full information about any remaining absolute differences in the specific column, so you will still be able to find any remaining small differences. 

My reason for asking Renato to round this is was that for me, the relative difference allows a quick visual check of relevance and to filter by "what is really important, and what can be ignored for the moment", and that is much easier if you have only full percentage values and not decimals after the dot.

Reply:
I just think it is confusing to have data that fails the checkSummation and then see `refdiff = 0`.

Reply:
but will you use this information for anything?
 I mean we know that when you round something below 0.5 you get 0, so I don't think this is very confusing :-)

I don't have a strong reason to change this, but I have wasted so much time by always having to click "change decimals to 0" when opening mifs that I feel this drive to optimize fast usability (as long as you can still get the full information based on the absolute difference)

Reply:
I suggest using the already exisiting roundDiff variable to decide if the csv output should be rounded.

https://github.com/Renato-Rodrigues/piamInterfaces/pull/1

Reply:
I am against using the `roundDiff ` alternative, as it affects the values that will be considered as deviations in the output.

Rounding the relative difference column will preserve the smaller deviations in the outputs, even if their relative difference is rounded.

Reply:
`roundDiff` is a boolean either switching round on or off

Reply:
I adjusted `niceround`, you can use it with `niceround(reldiff, digits = 1)`, so you get
- 12.41 -> 12
- 1.5 -> 1
- 0.432 -> 0.4

Reply:
> `roundDiff` is a boolean either switching rounding on or off (see my code in the PR)
> 
> It is written in a way that it does not affect later summaries, but only the format of the output csv

sorry for the misunderstanding then. I merged your PR.